gef-ui/server/controllers/bank-review-date/post-bank-review-date.ts

```diff
@@ -0,0 +1,161 @@
+import { Response } from 'express';
+import { isSameDay, parseISO } from 'date-fns';
+import { CustomExpressRequest } from '@ukef/dtfs2-common';
+import { isTrueSet, validationErrorHandler } from '../../utils/helpers';
+import * as api from '../../services/api';
+import { validateAndParseBankReviewDate } from './validation';
+import { asLoggedInUserSession, LoggedInUserSession } from '../../utils/express-session';
+import { getCoverStartDateOrStartOfToday } from '../../utils/get-cover-start-date-or-start-of-today';
+import { BankReviewDateViewModel } from '../../types/view-models/bank-review-date-view-model';
+import { Facility } from '../../types/facility';
+
+type BankReviewDateParams = { dealId: string; facilityId: string };
+type BankReviewDatePostBody = { 'bank-review-date-day': string; 'bank-review-date-month': string; 'bank-review-date-year': string };
+
+type PostBankReviewDateRequest = CustomExpressRequest<{
+  reqBody: BankReviewDatePostBody;
+  params: BankReviewDateParams;
+  query: { saveAndReturn: string; status: string | undefined };
+}>;
+
+type PostRequestUris = {
+  nextPage: string;
+  previousPage: string;
+  saveAndReturn: string;
+};
+
+type PostBankReviewDateParams = {
+  req: PostBankReviewDateRequest;
+  res: Response;
+  uris: PostRequestUris;
+};
```

Comment:
I can see there are duplicates of the above 5x types, can we commonise these / be DRY?

Reply:
These are different depending on being bank review date/ facility end date and post/get requests

Reply:
Sorry didn't notice - maybe could create a new type and extend it? Thinking something like this:

```ts
interface aNewType {
  res: Response;
  uris: PostRequestUris;  
}

interface HandlePostFacilityEndDateParams extends aNewType {
  req: PostFacilityEndDateRequest;  
}
```

Or would this be overkill / future PR ?

Reply:
bump

Reply:
My instinct is overkill since they're only used in 2 places & aren't really reusable for anything else